votor/src/consensus_pool_service.rs

```diff
@@ -412,3 +412,269 @@ impl ConsensusPoolService {
+            .rev()
+            .map(|i| ((i.saturating_add(5).saturating_mul(100)) as u64))
+            .collect::<Vec<_>>();
+        let genesis = create_genesis_config_with_alpenglow_vote_accounts(
+            1_000_000_000,
+            &validator_keypairs,
+            stake,
+        );
+        let my_keypair = validator_keypairs[0].node_keypair.insecure_clone();
+        let contact_info = ContactInfo::new_localhost(&my_keypair.pubkey(), 0);
+        let cluster_info = ClusterInfo::new(
+            contact_info,
+            Arc::new(my_keypair),
+            SocketAddrSpace::Unspecified,
+        );
+        let bank0 = Bank::new_for_tests(&genesis.genesis_config);
+        let bank_forks = BankForks::new_rw_arc(bank0);
+
+        let ledger_path = get_tmp_ledger_path_auto_delete!();
+        let blockstore = Blockstore::open(ledger_path.path()).unwrap();
+        let root_bank = bank_forks.read().unwrap().sharable_root_bank();
+        let exit = Arc::new(AtomicBool::new(false));
+        let leader_schedule_cache = Arc::new(LeaderScheduleCache::new_from_bank(&root_bank.load()));
+        let ctx = ConsensusPoolContext {
+            exit: exit.clone(),
+            start: Arc::new((Mutex::new(true), Condvar::new())),
+            cluster_info: Arc::new(cluster_info),
+            my_vote_pubkey: Pubkey::new_unique(),
+            blockstore: Arc::new(blockstore),
+            root_bank: root_bank.clone(),
+            leader_schedule_cache: leader_schedule_cache.clone(),
+            consensus_message_receiver,
+            bls_sender,
+            event_sender,
+            commitment_sender,
+        };
+        ConsensusPoolServiceTestComponents {
+            consensus_pool_service: ConsensusPoolService::new(ctx),
+            consensus_message_sender,
+            bls_receiver,
+            event_receiver,
+            commitment_receiver,
+            validator_keypairs,
+            exit,
+            leader_schedule_cache,
+            root_bank,
+        }
+    }
+
+    fn wait_for_event<T>(receiver: &Receiver<T>, condition: impl Fn(&T) -> bool) {
+        let start = Instant::now();
+        let mut event_received = false;
+        while start.elapsed() < Duration::from_secs(5) {
+            if let Ok(event) = receiver.recv_timeout(Duration::from_millis(500)) {
+                if condition(&event) {
+                    event_received = true;
+                    break;
+                }
+            }
+        }
+        assert!(event_received);
+    }
+
+    #[test]
+    fn test_receive_and_send_consensus_message() {
+        solana_logger::setup();
+        let setup_result = setup();
+        let consensus_message_sender = setup_result.consensus_message_sender;
+        let consensus_pool_service = setup_result.consensus_pool_service;
+
+        // validator 0 to 7 send Notarize on slot 2
+        let block_id = Hash::new_unique();
+        let notarize_vote = Vote::new_notarization_vote(2, block_id);
```

Comment:
For readability, could we define a variable `target_slot = 2` to avoid magic numbers

Reply:
Done.